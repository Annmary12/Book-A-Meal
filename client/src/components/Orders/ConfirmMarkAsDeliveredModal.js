import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import styles from './order.scss';

const ConfirmMarkAsDeliveredModal = ({
  orderId, isOpen, handleClose, deliverOrder
}) => (
  <ReactModal
    isOpen={isOpen}
    onRequestClose={handleClose}
    overlayClassName="react-modal-overlay"
    className="react-modal-body"
    closeTimeoutMS={150}
  >
    <div className={styles.body}>
      <p>Are you sure you want to mark order as delivered?</p>
      <button
        className={styles.successBtn}
        onClick={() => {
          deliverOrder(orderId);
          }}
      >
        Yes
      </button>
      <button
        className={styles.updateBtn}
        onClick={handleClose}
      >
        No
      </button>
    </div>
  </ReactModal>
);

ConfirmMarkAsDeliveredModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  deliverOrder: PropTypes.func.isRequired,
  orderId: PropTypes.string.isRequired
};

export default ConfirmMarkAsDeliveredModal;
