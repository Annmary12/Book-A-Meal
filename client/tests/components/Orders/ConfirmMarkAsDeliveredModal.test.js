import React from 'react';
import ConfirmMarkAsDeliveredModal from '../../../src/components/Orders/ConfirmMarkAsDeliveredModal';

const props = {
  isOpen: false,
  handleClose: jest.fn(),
  deliverOrder: jest.fn(),
  orderId: 'dae2dfsdf'
}
describe('ConfirmMarkAsDeliveredModal component', () => {
  const wrapper = shallow(
    <ConfirmMarkAsDeliveredModal {...props} />
  );

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('deliverOrder method', () => {
    it('should call deliverOrder prop function with orderId from prop', () => {
      wrapper.instance().deliverOrder();

      expect(props.deliverOrder).toBeCalledWith(props.orderId);
    })
  });
});