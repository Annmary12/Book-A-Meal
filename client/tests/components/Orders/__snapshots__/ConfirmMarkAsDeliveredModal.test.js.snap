// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmMarkAsDeliveredModal component should match snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="react-modal-body"
  closeTimeoutMS={150}
  isOpen={false}
  onRequestClose={[MockFunction]}
  overlayClassName="react-modal-overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="body"
  >
    <p>
      Are you sure you want to mark order as delivered?
    </p>
    <button
      className="successBtn"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="updateBtn"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</Modal>
`;
